browser/js/items/item_page.js

```diff
@@ -8,12 +8,18 @@ app.config(function($stateProvider) {
 	});
 });
 
-app.controller('ItemCtrl', function($scope, $state, $stateParams, SingleItemFactory) {
+app.controller('ItemCtrl', function($scope, $state, $stateParams, SingleItemFactory, CurrentFactory) {
 	SingleItemFactory.getItem($stateParams.productId).then(function (data){
 		$scope.item = data;
-		$scope.showDescription = $scope.item.description !== "";
-		SingleItemFactory.getUser(data.sellerID).then(function (username){
-			$scope.item.sellerName = username;
+		$scope.showDescription = $scope.item.description != "";
+		SingleItemFactory.getUser(data._id, data.sellerID).then(function (user){
```

Comment:
Can you use the UserFactory getById here? YOu just need to pass in data.sellerID.